Mathlib/Logic/Basic.lean

```diff
@@ -115,10 +114,12 @@ section Propositional
 alias Iff.imp := imp_congr
 
 -- This is a duplicate of `Classical.imp_iff_right_iff`. Deprecate?
-theorem imp_iff_right_iff {a b : Prop} : (a → b ↔ b) ↔ a ∨ b := Decidable.imp_iff_right_iff
+theorem imp_iff_right_iff {a b : Prop} : (a → b ↔ b) ↔ a ∨ b :=
+  open scoped Classical in Decidable.imp_iff_right_iff
```

Comment:
Is this pattern preferred to
```suggestion
  by classical exact Decidable.imp_iff_right_iff
```
?

Reply:
I don't feel strongly either way, I wrote things the way I did to preserve the term-mode / tactic-mode choice previously made.

Reply:
I tend to prefer `by classical exact` for this

Reply:
[#mathlib4 > &#96;by classical exact foo&#96; vs &#96;open scoped Classical in foo&#96; @ 💬](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/.60by.20classical.20exact.20foo.60.20vs.20.60open.20scoped.20Classical.20in.20foo.60/near/510632814)

Reply:
Agreed!